Add cluster protocol

cyrilwanner opened the pull request:
This describes the protocol within a cluster of nodes to get some feedback before implementation.

Please check if that makes sense to you or if you find any possible improvement or use-case I didn't think about. Thanks!

ItsEcholot reviewed (commented):
I think the specifications looks good and contains all the things we need (for the foreseeable future). But if I remember correctly ProtoBuf doesn't contain a method to detect what message is actually being sent, which is needed to decode the message correctly. Thus I would propose the following:

Creating a _wrapper_ message which could encapsulate all other message types and contains what message is actually sent and a few other bits like the magic number and maybe additionally a protocol version number (not sure if this is needed in our case)?

It could look like this:
```protobuf
message Wrapper {
    required uint32 messageType = 1;
    oneof msg {
        ServiceAnnouncement serviceAnnouncement = 2;
        ServiceAcquisition serviceAcquisition = 3;
        ...
    }   
}
```

cyrilwanner commented:
Thank you, I did not think about that but it makes total sense!
I added the wrapper as you have suggested.

Protobuffs are generally forward- and backward-compatible when implemented and changed correctly, so I don't think that the `version` is required, but I still added it since it can't hurt to have it. And we are then able to display a node as outdated in the GUI while still allowing it in the cluster (if there wasn't a breaking change in the protocol).

ItsEcholot reviewed (approved):
Perfect, thank you 👍
